main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -1560,11 +1560,11 @@ object Typer {
         } yield (constrs.flatten, resultTyp, resultPur, resultEff)
 
 
-      case KindedAst.Expression.NewChannel(exp, declaredType, loc) =>
+      case KindedAst.Expression.NewChannel(exp, loc) =>
         for {
           (constrs, tpe, _, eff) <- visitExp(exp)
           _ <- expectTypeM(expected = Type.Int32, actual = tpe, exp.loc)
-          resultTyp <- liftM(Type.mkChannel(declaredType, loc))
+          resultTyp <- liftM(Type.mkChannel(tpe, loc))
```

Comment:
Try `Type.mkChannel(tpe, loc)` -> `Type.mkFreshVar()` (or whatever its called).

The problem is that you are saying the type of the channel is the type of the expression (but the expression is of type int).

Reply:
Ah! Right...